service-monitor/src/main/java/com/yahoo/vespa/service/model/ApplicationInstanceGenerator.java

```diff
@@ -53,8 +55,21 @@ public ApplicationInstance makeApplicationInstance(ServiceStatusProvider service
 
         for (HostInfo host : applicationInfo.getModel().getHosts()) {
             HostName hostName = new HostName(host.getHostname());
+
+            boolean isTenantHost =
+                    applicationInfo.getApplicationId().equals(ZoneApplication.getApplicationId()) &&
+                    host.getServices().stream().anyMatch(serviceInfo ->
+                            ZoneApplication.isNodeAdminServiceInfo(applicationInfo.getApplicationId(), serviceInfo));
+
```

Comment:
For readability, you could consider extracting the last two lines to a method called e.g. 'hasNodeAdminService'